java/client/src/main/java/glide/api/models/commands/WeightAggregateOptions.java

```diff
@@ -0,0 +1,68 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import glide.api.commands.SortedSetBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.Singular;
+
+/**
+ * Optional arguments to {@link SortedSetBaseCommands#zunionstore(String, String[],
+ * WeightAggregateOptions)}
+ *
+ * @see <a href="https://redis.io/commands/zunionstore/">redis.io</a> for more details.
+ */
+@Builder
+public final class WeightAggregateOptions {
+    public static final String WEIGHTS_REDIS_API = "WEIGHTS";
+    public static final String AGGREGATE_REDIS_API = "AGGREGATE";
+
+    /**
+     * Represents multiplication factors for each sorted set, ready for aggregation. Each
+     * multiplication factor corresponds one-to-one with the sets. The score of every element in these
+     * sets is multiplied by its associated factor before aggregation.
+     */
+    @Singular private final List<Double> weights;
+
+    /**
+     * Option for the method of aggregating scores from multiple sets. This option defaults to SUM if
+     * not specified.
+     */
+    private final Aggregate aggregate;
+
+    /**
+     * Option for the method of aggregating scores from multiple sets. This option defaults to SUM if
+     * not specified.
+     */
+    public enum Aggregate {
+        /** Aggregates by summing the scores of each element across sets. */
+        SUM,
```

Comment:
nit: consider including the redisApi constructor with a constant string for output (like you have for `WEIGHTS_REDIS_API`).  That way you don't need to include hardcoded strings in your tests for "SUM", "MIN", etc.

Reply:
No need you can just call to string on the enum itself will fix where appropriate.